cpp/src/community/edge_triangle_count_impl.cuh

```diff
@@ -78,77 +78,297 @@ struct update_edges_p_r_q_r_num_triangles {
+          edge_first});
+
+      thrust::sort(handle.get_thrust_policy(),
+                   get_dataframe_buffer_begin(vertex_pair_buffer_tmp),
+                   get_dataframe_buffer_end(vertex_pair_buffer_tmp));
+
+      rmm::device_uvector<edge_t> increase_count_tmp(2 * intersection_indices.size(),
+                                                     handle.get_stream());
+      thrust::fill(handle.get_thrust_policy(),
+                   increase_count_tmp.begin(),
+                   increase_count_tmp.end(),
+                   size_t{1});
+
+      auto count_p_r_q_r = thrust::unique_count(handle.get_thrust_policy(),
+                                                get_dataframe_buffer_begin(vertex_pair_buffer_tmp),
+                                                get_dataframe_buffer_end(vertex_pair_buffer_tmp));
+
+      rmm::device_uvector<edge_t> increase_count(count_p_r_q_r, handle.get_stream());
+
+      auto vertex_pair_buffer = allocate_dataframe_buffer<thrust::tuple<vertex_t, vertex_t>>(
+        count_p_r_q_r, handle.get_stream());
+      thrust::reduce_by_key(handle.get_thrust_policy(),
+                            get_dataframe_buffer_begin(vertex_pair_buffer_tmp),
+                            get_dataframe_buffer_end(vertex_pair_buffer_tmp),
+                            increase_count_tmp.begin(),
+                            get_dataframe_buffer_begin(vertex_pair_buffer),
+                            increase_count.begin(),
+                            thrust::equal_to<thrust::tuple<vertex_t, vertex_t>>{});
+
+      rmm::device_uvector<vertex_t> pair_srcs(0, handle.get_stream());
+      rmm::device_uvector<vertex_t> pair_dsts(0, handle.get_stream());
+      std::optional<rmm::device_uvector<edge_t>> pair_count{std::nullopt};
+
+      std::optional<rmm::device_uvector<edge_t>> opt_increase_count =
+        std::make_optional(rmm::device_uvector<edge_t>(increase_count.size(), handle.get_stream()));
+
+      raft::copy<edge_t>((*opt_increase_count).begin(),
+                         increase_count.begin(),
+                         increase_count.size(),
+                         handle.get_stream());
+
+      // There are still multiple copies here but is it worth sorting and reducing again?
+      std::tie(pair_srcs, pair_dsts, std::ignore, pair_count, std::ignore) =
+        shuffle_int_vertex_pairs_with_values_to_local_gpu_by_edge_partitioning<vertex_t,
+                                                                               edge_t,
+                                                                               weight_t,
+                                                                               int32_t>(
+          handle,
+          std::move(std::get<0>(vertex_pair_buffer)),
+          std::move(std::get<1>(vertex_pair_buffer)),
+          std::nullopt,
+          // FIXME: Update 'shuffle_int_...' to support int32_t and int64_t values
```

Comment:
>  You can use int32_t and int64_t values as you are currently doing this.

Right. This is currently how we proceed but based on Seunghwa feedbacks on `decompress_edgelist,` we should have something separate from `edge_id` to handle this in case `edge_id` and count don't hold the same type. I thought the same holds for the shuffling function.

Reply:
I guess I'm asking for a more descriptive FIXME.  As worded it is not clear to me what you are asking to be changed at some point in the future.

Reply:
Right. I updated the FIXME and it will be available in the next commit

Reply:
Done